oneflow/api/cpp/framework/graph.cpp

```diff
@@ -0,0 +1,353 @@
+      } else if (op_conf.has_variable_conf()) {
+        const of::LazyMode::Guard lazy_mode_disabled_guard{false};
+        const of::VariableOpConf variable_conf = op_conf.variable_conf();
+        variable_op_name_to_tensor_[op_conf.name()] = JUST(of::one::functional::Empty(
+            of::Shape(variable_conf.shape()),
+            JUST(of::DType::Get(static_cast<of::DataType>(variable_conf.data_type()))),
+            *device_.device_));
+      }
+      return of::Maybe<void>::Ok();
+    }));
+  }
+  JUST(of::CurJobBuildAndInferCtx_Complete());
+  {
+    const std::shared_ptr<of::Job> complete_job = JUST(of::GetCurrentJob());
+    const of::OpGraph complete_graph(*complete_job);
+    JUST(complete_graph.ForEachOpNode([&](const of::OpNode& node) -> of::Maybe<void> {
+      const of::LazyMode::Guard lazy_mode_disabled_guard{false};
+      const of::OperatorConf& op_conf = node.op().op_conf();
+      if (op_conf.has_output_conf()) {
+        of::InterfaceBlobConf blob_conf = op_conf.output_conf().blob_conf();
+        if (batch_size_ > 0) {
+          const std::string input_lbi_str = op_conf.output_conf().in();
+          const of::LogicalBlobId input_lbi = of::GenLogicalBlobId(input_lbi_str);
+          int64_t batch_size = node.LogicalBlobDesc4Lbi(input_lbi).shape().At(0);
+          blob_conf.mutable_shape()->set_dim(0, batch_size);
+        }
+        output_name_to_tensor_[op_conf.name()] = JUST(of::one::functional::Empty(
+            of::Shape(blob_conf.shape()),
+            JUST(of::DType::Get(static_cast<of::DataType>(blob_conf.data_type()))),
+            *device_.device_));
+      }
+      return of::Maybe<void>::Ok();
+    }));
+  }
+  return of::Maybe<void>::Ok();
+}
+
+of::Maybe<void> Graph::GraphImpl::LoadCheckpoint() {
+  for (const auto& variable_op_name_and_tensor : variable_op_name_to_tensor_) {
+    const auto& variable_op_name = variable_op_name_and_tensor.first;
+    const auto& variable_tensor = variable_op_name_and_tensor.second;
+    const std::string variable_filename = model_path_ + "/" + variable_op_name + "/out";
+    const std::string buffer = [&variable_filename]() {
+      std::ifstream variable_file(variable_filename, std::ios::binary);
+      CHECK(variable_file.is_open());
+      std::stringstream ss;
+      ss << variable_file.rdbuf();
+      return ss.str();
+    }();
+    const auto& callback =
+        std::make_shared<std::function<void(uint64_t)>>([&](uint64_t of_blob_ptr) {
+          CHECK_JUST(of::BlobBufferCopyUtil<void>::From(
+              of_blob_ptr, buffer.data(),
+              variable_tensor->shape()->elem_cnt()
+                  * of::GetSizeOfDataType(variable_tensor->dtype()->data_type())));
+        });
+    JUST(of::one::SyncAccessTensorWithTimeOut(variable_tensor, callback, "mut"));
+  }
+
+  return of::Maybe<void>::Ok();
+}
+
+of::Maybe<void> Graph::GraphImpl::RegisterTensors() {
+  {
+    const auto pair = Unzip(input_name_to_tensor_);
```

Comment:
```suggestion
    const auto& pair = Unzip(input_name_to_tensor_);
```

可以总是用 const auto& 而不考虑 Unzip 是返回值还是引用，常量引用会延长临时值的生命周期